### PR TITLE
Handle multi-resource YAMLs in github channel synchronizer

### PR DESCRIPTION
**Pull request by rokej**

### CODE REVIEW

**Comment by rokej** [comment]
https://github.com/open-cluster-management/backlog/issues/1247

File: pkg/utils/github.go [hunk 1]
```diff
@@ -218,3 +223,32 @@ func generateIndexYAML(repoRoot string) (*repo.IndexFile, map[string]string, err
 
 	return indexFile, resourceDirs, nil
 }
+
+// ParseKubeResoures parses a YAML content and returns kube resources in byte array from the file
+func ParseKubeResoures(file []byte) [][]byte {
```
Comment (tengqm) [inline pkg/utils/github.go]: https://github.com/kubernetes/apimachinery/blob/master/pkg/util/yaml/decoder.go ?

Comment (rokej) [reply pkg/utils/github.go]: Thanks for reviewing. The code I wrote suites our purpose better.

File: pkg/synchronizer/githubsynchronizer/githubsynchronizer.go [hunk 2]
```diff
@@ -158,19 +158,32 @@ func (sync *ChannelSynchronizer) processYamlFile(chn *chv1.Channel, files []os.F
...
-				if t.APIVersion == "" || t.Kind == "" {
-					klog.V(debugLevel).Info("Not a Kubernetes resource")
-				} else if err := sync.handleSingleDeployable(chn, dir, f, file, t, newDplList); err != nil {
-					klog.Errorf("Failed to handle single deployable, err: %+v", err)
+				resources := utils.ParseKubeResoures(file)
+
+				if len(resources) == 0 {
+					klog.Info("The file contains no Kubernetes resource.")
+					continue
+				} else {
```
Comment (ianzhang366) [inline pkg/synchronizer/githubsynchronizer/githubsynchronizer.go]: I feel we can drop this `else`, since the `if` has a `continue` statement.

Comment (rokej) [reply pkg/synchronizer/githubsynchronizer/githubsynchronizer.go]: Thanks for pointing that out. In this case, `continue` statement is useless so I am removing that.

**Pull request merged**